Lib/dbm/__init__.py

```diff
@@ -75,6 +75,7 @@ def open(file, flag='r', mode=0o666):
             raise ImportError("no dbm clone found; tried %s" % _names)
 
     # guess the type of an existing database, if not creating a new one
+    file = os.fsencode(file)
```

Comment:
Why was it added? Is it needed if `whichdb()` and `mod.open()` support path-like objects?